site-permissions/site-permissions-impl/src/main/java/com/duckduckgo/site/permissions/impl/SitePermissionsDialogActivityLauncher.kt

```diff
@@ -254,18 +262,23 @@ class SitePermissionsDialogActivityLauncher @Inject constructor(
             SitePermissionsRequestedType.AUDIO -> R.string.systemPermissionDialogAudioDeniedContent
             SitePermissionsRequestedType.CAMERA_AND_AUDIO -> R.string.systemPermissionDialogCameraAndAudioDeniedContent
         }
-        AlertDialog.Builder(activity).apply {
-            setTitle(titleRes)
-            setMessage(contentRes)
-            setPositiveButton(R.string.systemPermissionsDeniedDialogPositiveButton) { _, _ ->
-                val intent = Intent()
-                intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
-                val uri = Uri.fromParts("package", activity.packageName, null)
-                intent.data = uri
-                context.startActivity(intent)
-            }
-            setNegativeButton(R.string.systemPermissionsDeniedDialogNegativeButton) { _, _ -> }
-            show()
+        (activity as FragmentActivity?)?.supportFragmentManager?. let { fragmentManager ->
+            TextAlertDialog.Builder(activity)
+                .setTitle(titleRes)
+                .setMessage(contentRes)
+                .setPositiveButton(R.string.systemPermissionsDeniedDialogPositiveButton)
+                .setNegativeButton(R.string.systemPermissionsDeniedDialogNegativeButton)
+                .addEventListener(object : EventListener() {
+                    override fun onPositiveButtonClicked() {
+                        val intent = Intent()
+                        intent.action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+                        val uri = Uri.fromParts("package", activity.packageName, null)
+                        intent.data = uri
+                        activity.startActivity(intent)
```

Comment:
```suggestion
                        context.startActivity(intent)
```

Reply:
You are right! I was using it before this change, not sure why I changed it. Thanks! 💯